File: Jack2/src/main/java/ch/chassaing/jack/token/Symbol.java
package ch.chassaing.jack.token;

public enum Symbol
    implements Token
{
    LEFT_BRACE('{'), RIGHT_BRACE('}'),
    LEFT_PAREN('('), RIGHT_PAREN(')'),
    LEFT_BRACKET('['), RIGHT_BRACKET(']'),
    DOT('.'), COMMA(','), SEMICOLON(';'),
    PLUS('+'), MINUS('-'),
    STAR('*'), SLASH('/'),
    AMP('&'), PIPE('|'),
    LT('<'), GT('>'), EQUAL('='),
    NEG('~');

    public final char repr;

    Symbol(char repr) {this.repr = repr;}
}

File: JackCompiler/src/main/kotlin/ch/chassaing/jack/Symbol.kt
package ch.chassaing.jack

enum class Symbol(val c : Char)  {

    LEFT_BRACE('{'), RIGHT_BRACE('}'),
    LEFT_PAREN('('), RIGHT_PAREN(')'),
    LEFT_BRAKET('['), RIGHT_BRAKET(']'),
    DOT('.'), COMMA(','), SEMICOLON(';'),
    PLUS('+'), MINUS('-'),
    STAR('*'), SLASH('/'),
    AMP('&'), PIPE('|'),
    LT('<'), GT('>'), EQUAL('='),
    NEG('~');
}